app/src/main/java/org/phenoapps/verify/MainActivity.java

```diff
@@ -563,78 +562,76 @@ private synchronized void askUserExportFileName() {
         if (lastDot != -1) {
             mFileName = mFileName.substring(0, lastDot);
         }
-        input.setText(mFileName + "_" + sdf.format(c.getTime()));
-        input.setInputType(InputType.TYPE_CLASS_TEXT);
-        builder.setView(input);
 
-        builder.setPositiveButton("Export", new DialogInterface.OnClickListener() {
-            @Override
-            public void onClick(DialogInterface dialog, int which) {
-                String value = input.getText().toString();
-                if (!value.isEmpty()) {
-                    if (isExternalStorageWritable()) {
-                        try {
-                            File verifyDirectory = new File(Environment.getExternalStorageDirectory().getPath() + "/Verify");
-                            final File output = new File(verifyDirectory, value + ".csv");
-                            final FileOutputStream fstream = new FileOutputStream(output);
-                            final SQLiteDatabase db = mDbHelper.getReadableDatabase();
-                            final String table = IdEntryContract.IdEntry.TABLE_NAME;
-                            final Cursor cursor = db.query(table, null, null, null, null, null, null);
-                            //final Cursor cursor = db.rawQuery("SElECT * FROM VERIFY", null);
-
-                            //first write header line
-                            final String[] headers = cursor.getColumnNames();
+        final Intent i = new Intent(Intent.ACTION_CREATE_DOCUMENT);
+        i.setType("*/*");
+        i.putExtra(Intent.EXTRA_TITLE, mFileName + "_" + sdf.format(c.getTime())+".csv");
+        startActivityForResult(Intent.createChooser(i, "Choose folder to export file."), VerifyConstants.PICK_CUSTOM_DEST);
+
+    }
+
+    public void writeToExportPath(Uri uri){
+
+        String value = mFileName;
+
+        if (!value.isEmpty()) {
+            if (isExternalStorageWritable()) {
+                try {
+                    final File output = new File(uri.getPath());
+                    final OutputStream fstream = getContentResolver().openOutputStream(uri);
```

Comment:
fstream can be null, same as above, catch this possible NPE and show an error message